src/components/Article-Components/SingleArticle.jsx

```diff
@@ -22,26 +22,43 @@ const SingleArticle = () => {
             })
     }, [article_id])
 
-    if (isLoading) {
-        return <p> Loading.... </p>
-    }
 
-    if (article_id) {
-        return (
-            <>
-                <IndividualArticleCard
-                    title={singleArticle.title}
-                    author={singleArticle.author}
-                    body={singleArticle.body}
-                    topic={singleArticle.topic}
-                    comments={singleArticle.comments}
-                    published={new Date(singleArticle.created_at).toUTCString()}
-                    img=<img src={singleArticle.article_img_url} alt="article image" width="100" height="100" />
-                />
-            </>
-        )
+    const updateVotes = (article_id, increment) => {
+        addArticleVote(article_id, increment)
+            .then((updatedArticle) => {
+                setSingleArticle((article) => { return { ...article, votes: updatedArticle.votes } })
+            })
+            .catch((err) => {
```

Comment:
Nice. However, we want optimistic rendering here. So you want to increase the votes on click, then make the API request, then undo the optimistic update if the catch block is reached.